File: Colaveco/JavaModel/mobile/Android/gxcommon/com/colaveco/loger.java
/*
               File: Loger
        Description: Loger
             Author: GeneXus Android Generator version 17_0_7-156546
       Generated on: November 14, 2022 10:7:19.30
       Program type: Callable routine
          Main DBMS: SQLite
*/
package com.colaveco ;
import com.colaveco.*;
import com.genexus.*;
import com.genexus.db.*;
import java.sql.*;
import com.artech.base.services.*;

public final  class loger extends GXProcedure implements IGxProcedure
{
   public boolean execute( IPropertiesObject androidPropertiesObject )
   {

      String aP0 = "";

      aP0 = (String) androidPropertiesObject.optStringProperty("LogDetalle");

      execute(aP0);



      return true ;
   }

   public loger( int remoteHandle )
   {
      super( remoteHandle , new ModelContext( loger.class ), "" );
   }

   public loger( int remoteHandle ,
                 ModelContext context )
   {
      super( remoteHandle , context, "" );
   }

   public void execute( String aP0 )
   {
      execute_int(aP0);
   }

   private void execute_int( String aP0 )
   {
      loger.this.AV8LogDetalle = aP0;
      initialize();
      /* GeneXus formulas */
      /* Output device settings */
      privateExecute();
   }

   private void privateExecute( )
   {
      /*
         INSERT RECORD ON TABLE Log

      */
      A174LogDetalle = AV8LogDetalle ;
      n174LogDetalle = false ;
      /* Using cursor P00102 */
      pr_default.execute(0, new Object[] {new Boolean(n174LogDetalle), A174LogDetalle});
      /* Retrieving last key number assigned */
      /* Using cursor P00103 */
      pr_default.execute(1);
      A175LogId = P00103_A175LogId[0] ;
      pr_default.close(1);
      if ( (pr_default.getStatus(0) == 1) )
      {
         Gx_err = (short)(1) ;
         Gx_emsg = localUtil.getMessages().getMessage("GXM_noupdate") ;
      }
      else
      {
         Gx_err = (short)(0) ;
         Gx_emsg = "" ;
      }
      /* End Insert */
      cleanup();
   }

   protected void cleanup( )
   {
      Application.commitDataStores(context, remoteHandle, pr_default, "loger");
      CloseOpenCursors();
      exitApplication();
   }

   protected void CloseOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public void initialize( )
   {
      A174LogDetalle = "" ;
      scmdbuf = "" ;
      P00103_A175LogId = new long[1] ;
      Gx_emsg = "" ;
      pr_default = new DataStoreProvider(context, remoteHandle, new com.colaveco.loger__default(),
         new Object[] {
             new Object[] {
            }
            , new Object[] {
            P00103_A175LogId
            }
         }
      );
      /* GeneXus formulas. */
      Gx_err = (short)(0) ;
   }

   private short Gx_err ;
   private int GX_INS7 ;
   private long A175LogId ;
   private String scmdbuf ;
   private String Gx_emsg ;
   private boolean n174LogDetalle ;
   private String AV8LogDetalle ;
   private String A174LogDetalle ;
   private IDataStoreProvider pr_default ;
   private long[] P00103_A175LogId ;
}

final  class loger__default extends DataStoreHelperBase implements ILocalDataStoreHelper
{
   public Cursor[] getCursors( )
   {
      return new Cursor[] {
          new UpdateCursor("P00102", "INSERT INTO [Log]([LogDetalle]) VALUES(?)", GX_NOMASK + GX_MASKLOOPLOCK)
         ,new ForEachCursor("P00103", "SELECT last_insert_rowid() FROM [Log] ",false, GX_NOMASK + GX_MASKLOOPLOCK, false, this,1, 0,false )
      };
   }

   public void getResults( int cursor ,
                           IFieldGetter rslt ,
                           Object[] buf ) throws SQLException
   {
      switch ( cursor )
      {
            case 1 :
               ((long[]) buf[0])[0] = rslt.getLong(1);
               return;
      }
   }

   public void setParameters( int cursor ,
                              IFieldSetter stmt ,
                              Object[] parms ) throws SQLException
   {
      switch ( cursor )
      {
            case 0 :
               if ( ((Boolean) parms[0]).booleanValue() )
               {
                  stmt.setNull( 1 , Types.LONGVARCHAR );
               }
               else
               {
                  stmt.setLongVarchar(1, (String)parms[1]);
               }
               return;
      }
   }

}

File: Colaveco/JavaModel/mobile/iOS/gxcommon/loger.swift
/*
               File: Loger
        Description: Loger
             Author: GeneXus Swift Generator version 17_0_7-156546
       Generated on: November 14, 2022 10:7:19.69
       Program type: Callable routine
          Main DBMS: SQLite
*/

@objc( loger )
public final  class loger: GXProcedure
{
   @objc public func executeFlexibleClient( _ aP0: NSString? )
   {
      let iP0: String = (aP0 == nil) ? "" : aP0! as String
      execute(iP0)
   }

   public func execute( _ aP0: String )
   {
      execute_int(aP0)
   }

   private func execute_int( _ aP0: String )
   {
      self.AV8LogDetalle = aP0
      initialize()
      /* GeneXus formulas */
      /* Output device settings */
      autoreleasepool
      {
         privateExecute()
      }
   }

   private func privateExecute( )
   {
      /*
         INSERT RECORD ON TABLE Log

      */
      A174LogDetalle = AV8LogDetalle
      n174LogDetalle = false
      /* Using cursor P00102 */
      var params0 = Array<Any>()
      params0.append(A174LogDetalle)
      var nullParams0 = Array<Bool>()
      nullParams0.append(n174LogDetalle)
      pr_default.execute(0, params: params0, nullParams: nullParams0)
      /* Retrieving last key number assigned */
      /* Using cursor P00103 */
      pr_default.execute(1)
      self.A175LogId = pr_default.getDBCommand(1).int64Value(forColumn: "LogId")
      pr_default.close(1);
      if ( pr_default.getStatus(0) == 1 )
      {
         Gx_err = 1
         Gx_emsg = GXMessages.getMessage("GXM_noupdate")
      }
      else
      {
         Gx_err = 0
         Gx_emsg = ""
      }
      /* End Insert */
      cleanup()
   }

   public override func cleanup( )
   {
      GXDBManager.sharedInstance().commitDatastore("GAM")
      GXDBManager.sharedInstance().commitDatastore("DEFAULT")
      closeOpenCursors()
   }

   private func closeOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public override func initialize( )
   {
      A174LogDetalle = ""
      scmdbuf = ""
      Gx_emsg = ""
      pr_default = GXDataStoreProvider(loger__default(),
         [
         ]
      );
      /* GeneXus formulas. */
      Gx_err = 0
   }

   private var GX_INS7 : Int = 0
   private var Gx_err : Int = 0
   private var A175LogId : Int64 = 0
   private var scmdbuf : String = ""
   private var Gx_emsg : String = ""
   private var n174LogDetalle : Bool = false
   private var AV8LogDetalle : String = ""
   private var A174LogDetalle : String = ""
   private var pr_default : GXDataStoreProvider = GXDataStoreProvider()
}

@objc( loger__default )
final  class loger__default: GXDataStoreHelperBase
{
   override func getCursor( _ cursor: Int ) -> GXCursor?
   {
      switch cursor {
         case 0: return UpdateCursor("P00102", "INSERT INTO [Log]([LogDetalle]) VALUES(?)")
         case 1: return ForEachCursor("P00103", "SELECT last_insert_rowid() AS [LogId] FROM [Log] ")
         default: return nil
      }
   }

}

